kiva/fonttools/font.py

```diff
@@ -98,21 +193,35 @@ class Font(object):
     def __init__(self, face_name="", size=12, family=SWISS,
                  weight=WEIGHT_NORMAL, style=NORMAL, underline=0,
                  encoding=DEFAULT):
-        if (not isinstance(face_name, str)
-                or not isinstance(size, int)
-                or not isinstance(family, int)
-                or not isinstance(weight, int)
-                or not isinstance(style, int)
-                or not isinstance(underline, int)
-                or not isinstance(encoding, int)):
+        if not isinstance(face_name, str):
```

Comment:
Please could we have some documentation of the meaning of the parameters? Some of them aren't obvious (like encoding).

For `encoding`: what does this represent? The validation checks that it's an `int`, the error message says it should be a `bool`, and the default value `DEFAULT` is taken from the list of families. I'm a bit confused.

Reply:
I actually have no idea what `encoding` is used for.

Reply:
Also, I'm really not a fan of the error-checking that this class does - it's checking types, but a lot of the values are really supposed be restricted to a handful of constants.

It seems a bit pointless - likely better handled these days by static type checking or something.

Reply:
I think it's best to just not break existing code here and if we want to give a nicer interface for new code then that should be introduced.

Reply:
I've added #933 to track this.